File: php/features.php
<?php
declare(strict_types=1);

// PhishIn SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PhishInFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PhishInBaseFeature();
            case "test":
                return new PhishInTestFeature();
            default:
                return new PhishInBaseFeature();
        }
    }
}

File: py/features.py
# PhishIn SDK feature factory

from feature.base_feature import PhishInBaseFeature
from feature.test_feature import PhishInTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PhishInBaseFeature(),
        "test": lambda: PhishInTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
